eng/common/pipelines/templates/steps/credscan.yml

```diff
@@ -0,0 +1,39 @@
+parameters:
+  SuppressionFilePath: 'eng\CredScanSuppression.json'
+  BaselineFilePath: ''
+steps:
+- pwsh: |
+    New-Item -Path "$(Build.sourcesdirectory)" -Name "credscan.tsv"
+    if ("$(Build.Reason)" -eq 'PullRequest') {
+      New-Item -Path "$(Build.sourcesdirectory)" -Name "changedFiles" -ItemType "directory"
+      (git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only)
+      | ForEach-Object { Add-Content -Path "$(Build.SourcesDirectory)\credscan.tsv" -Value "$(Build.SourcesDirectory)\$_"}
```

Comment:
Use forward slashes to be cross-OS compatible.

```
      (git diff "origin/$(System.PullRequest.TargetBranch)" HEAD --name-only)
      | ForEach-Object { Add-Content -Path "$(Build.SourcesDirectory)/credscan.tsv" -Value "$(Build.SourcesDirectory)/$_"}
```

Reply:
Also I'm not sure whether hardcoding `origin` here is always going to work, but I can't think of any scenarios where it won't at the moment.

Reply:
We have an issue to make this change applicable to more cases.
https://github.com/Azure/azure-sdk-tools/issues/2531
Will address the problem in the above issue.